lib/avro/builder/rake/avro_generate_task.rb

```diff
@@ -0,0 +1,47 @@
+require 'rake/tasklib'
+require 'avro/builder'
+
+module Avro
+  module Builder
+    module Rake
+      class AvroGenerateTask < ::Rake::TaskLib
+
+        attr_accessor :name, :task_namespace, :task_desc, :root,
+                      :load_paths, :dependencies
+
+        def initialize(name: :generate, dependencies: [])
+          @name = name
+          @task_namespace = :avro
+          @task_desc = 'Generate Avro schema files from Avro::Builder DSL'
+          @load_paths = []
+          @root = "#{Rails.root}/avro/dsl" if defined?(Rails)
+          @dependencies = dependencies
+
+          yield self if block_given?
+
+          define
+        end
+
+        private
+
+        # Define the rake task
+        def define
+          namespace task_namespace do
+            desc task_desc
+            task(name.to_sym => dependencies) do
+              raise '"root" must be specified for Avro DSL files' unless root
+              Avro::Builder.add_load_path(*[root, load_paths].flatten)
```

Comment:
Do we need to set the load path here or can we expect it to already be set by the railtie (or another initializer)?